lib/max_subarray.py

```diff
@@ -2,11 +2,25 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
```

Comment:
✨ Your implementation works, however space complexity would be O(1). Whatever the length of the input, n, you are only ever taking up a constant amount of additional space through the two variables `max_so_far` and `max_ending_here`.